### ISSUE TITLE
CrossFade using Layers

### ISSUE THREAD

**Issue by Haytam95** [KybernetikGames/animancer#222]
Hi!

I'm using Animancer to handle my locomotion and custom animations. My setup is something like this:

![image](https://user-images.githubusercontent.com/7773130/185477515-390c76b4-0f6d-41df-8144-595d6084ab71.png)

So, when my "Pawn" (I called that) are in a Locomotion mode, "Normal locomotion" or "Crouched locomotion" layers change theirs weight. When the Pawns are required to perform a custom animation, it's added into a new layer called "Custom Animation" and when the animation is done, the Normal locomotion gets it's weight again.

It work nice, and it's very easy to make the Pawns perform custom animations. The problem is, that I don't know how I should crossfade the weight.

It only works nice whe going from ``Crouching Locomotion`` to ``Normal Locomotion``

Here is my code for normal / crouching:

```C#
// Walking
            if (_NormalLocomotion.Asset != null)
            {
                if (currentState == PawnStates.NORMAL || currentState == PawnStates.BUSY)
                {
                    _animancerComponent.Layers[layerNomalLocomotion].StartFade(1, 0.25f);
                    _animancerComponent.Layers[layerNomalLocomotion].Play(_NormalLocomotion, 0.25f);
                    if (_NormalLocomotion.State != null)
                    {
                        _NormalLocomotion.State.Parameter = currentVelocity.magnitude;
                    }
                }
                else
                {
                    _animancerComponent.Layers[layerNomalLocomotion].StartFade(0, 0.25f);
                }
            }


            // Crouching
            if (_CrouchingLocomotion.Asset != null)
            {
                if (currentState == PawnStates.CROUCH)
                {
                    _animancerComponent.Layers[layerCrouchingLocomotion].StartFade(1, 0.25f);
                    _animancerComponent.Layers[layerCrouchingLocomotion].Play(_CrouchingLocomotion, 0.25f);
                    if (_CrouchingLocomotion.State != null)
                    {
                        _CrouchingLocomotion.State.Parameter = currentVelocity.magnitude;
                    }
                }
                else
                {
                    _animancerComponent.Layers[layerCrouchingLocomotion].StartFade(0, 0.25f);
                }
            }
```

And here is my code to request a custom animation to play:

```C#
public AnimancerState PlayAnimation(AnimationClip clip)
{
            _animancerComponent.Layers[layerCustomAnimations].Stop();
            var state = _animancerComponent.Layers[layerCustomAnimations].Play(clip, 0.25f);
            state.Events.OnEnd += () => _animancerComponent.Layers[layerCustomAnimations].Stop();

            return state;
}
```

**Comment by KybernetikGames** [comment]
> The problem is, that I don't know how I should crossfade the weight.

What weight? It looks like you're using cross fading for all the transitions except for the custom animation. If you want that to fade out too, just fade out the layer like you are for the others instead of calling Stop.

**Comment by Haytam95** [comment]
You're right, i wasn't Cross Fading my custom animations.

And also I've found out an error with my current fade between locomotions, that error caused the animations to continue playing (even when the layer has weight 0), so when I called Play on them, the Fade would be ignored.

![image](https://user-images.githubusercontent.com/7773130/185511274-82dbf610-a9f8-46cd-b925-f2e1e6845f86.png)

Here is how I found out using the inspector.

For anyone that could need it: In my code, I just left the ``StartFade(0, [time])`` method to disable the layer, and left the ``Play(state)`` action to update the Layer and the State weight.

Thanks @KybernetikGames
